recipes/aws-c-cal/all/conanfile.py

```diff
@@ -98,4 +98,14 @@ def package_info(self):
         elif self.settings.os in ("FreeBSD", "Linux"):
             self.cpp_info.components["aws-c-cal-lib"].system_libs.append("dl")
         if self._needs_openssl:
-            self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
+            if self.options["openssl"].shared:
+                self.cpp_info.components["aws-c-cal-lib"].requires.append("openssl::crypto")
+            else:
+                self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
+                lib_path = os.path.join(self.deps_cpp_info["openssl"].rootpath, "lib", "libcrypto.a")
+                link_flag = "-Wl,--whole-archive,{},--no-whole-archive".format(lib_path)
+                self.cpp_info.components["aws-c-cal-lib"].exelinkflags.append(link_flag)
+                self.cpp_info.components["aws-c-cal-lib"].sharedlinkflags.append(link_flag)
+                self.cpp_info.components["aws-c-cal-lib"].system_libs.extend(["dl", "rt"])
+                if not self.options["openssl"].no_threads:
+                    self.cpp_info.components["aws-c-cal-lib"].system_libs.append("pthread")
```

Comment:
Using `--whole-archive` will pull in the complete crypto library and make the executable/library bigger then needed.
I propose the following alternative:
```suggestion
                # aws-c-cal dynamically search the openssl symbols.
                # Mark them as undefined so the linker will include them.
                # This avoid dynamical look-up for a system crypto library.
                crypto_symbols = [
                    "HMAC_CTX_init", "HMAC_CTX_cleanup", "HMAC_Update", "HMAC_Final", "HMAC_Init_ex",
                ]
                for crypto_symbol in cryto_symbols:
                    link_flag = f"-Wl,-u{crypto_symbol}"
                    self.cpp_info.components["aws-c-cal-lib"].exelinkflags.append(link_flag)
                    self.cpp_info.components["aws-c-cal-lib"].sharedlinkflags.append(link_flag)
```

Reply:
I think this approach can even be upstreamed (=added to their cmake script).

Reply:
I can't find your solution.
It is very clear way!
Thanks.

Reply:
I made some spelling fixes to my original suggestion, can you apply them as well?